pkg/storegateway/stats.go

```diff
@@ -172,6 +176,14 @@ func (s queryStats) merge(o *queryStats) *queryStats {
 	s.streamingSeriesIndexHeaderLoadDuration += o.streamingSeriesIndexHeaderLoadDuration
 	s.streamingSeriesConcurrencyLimitWaitDuration += o.streamingSeriesConcurrencyLimitWaitDuration
 
+	// For timestamps, take the latest one (not sum). This should only be set once for each request, so merging doesn't make a lot of sense anyways.
```

Comment:
If these should only be set once for each request, should we panic if it ever happens that they're set multiple times?